http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java

```diff
@@ -84,9 +87,10 @@ public static FeignClientFactory of(
             Encoder encoder,
             Decoder decoder,
             ErrorDecoder errorDecoder,
-            Function<Optional<SSLSocketFactory>, Client> clientSupplier) {
+            Function<Optional<SSLSocketFactory>, Client> clientSupplier,
+            String userAgent) {
```

Comment:
I think we need to release a few versions with both forms and `@Deprecated` the old one. otherwise upgrading may be very tedious, in particular with diamond-dependencies.